### PR TITLE
CODEOWNERS file

### PR DESCRIPTION
**Pull request by coot**

### CODE REVIEW

File: CODEOWNERS
Comment (nfrisby) [inline CODEOWNERS:18]: spelling: `ouroboros-netework`

File: CODEOWNERS
Comment (nfrisby) [inline CODEOWNERS:10]: spelling `netowrk-mux`

**Comment by nfrisby** [comment]
For future reference, here is the CODEOWNERS doc

https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners

and note that `cardano-ledger-specs` is already using one, so we can look at their example.

**Comment by coot** [comment]
@nfrisby I fixed typos and simplified it.

**Comment by nfrisby** [comment]
Ah ha, I found this in the docs.

> Order is important; the last matching pattern takes the most precedence.

**Comment by nfrisby** [comment]
I couldn't find sufficient documentation, so I ran an experiment on a throwaway repo with some of my friends. My main conclusion is: if someone is a code owner for a file then they will be a default reviewer for a PR that touches that file. You can't have owner permissions but not be a default reviewer. I tried the following two.

```
* @nfrisby
special_file @friend

# In this case, the PR on special_file had @friend as a reviewer by default, but me approving it wasn't enough to allow merge.
```

```
* @nfrisby
special_file @friend @nfrisby

# In this case, the PR on special_file had both @friend and me as reviewers by default.
```

So `CODEOWNERS` doesn't seem to have the expressivity I'd hoped for. EG I still want @dcoutts to be a sufficient Approval for any PR without every PR defaulting to including him as a reviewer. My motivation for this sort of thing is to reflect the decision hierarchy and also to avoid extra work to overcome potential delays due to my relative unavailability due to timezones.

---

The only angle I didn't pursue was the use of groups. For example, we can create a group that includes only Duncan but that he disables all notifications for. It'll be a default reviewer on all PRs -- so he can approve them -- but that will lead to no noise in his inbox.

@coot @dcoutts Thoughts?

**Comment by nfrisby** [comment]
OK, I took a pass at it. Now I'm seeing these files matched by no pattern except the default `*`.

```
$ git rev-parse HEAD
f32389451c889fb7c774bda6ba37f7c263b645ae
$ git status -suno
$ git ls-files | grep -Ev $(for i in $(cat .github/CODEOWNERS | grep -Ev -e ^\# -e '^$' -e '\*' | awk '{print $1}'); do echo "-e ^$i"; done)
.buildkite/nightly.yml
.buildkite/pipeline.yml
.gitattributes
.github/CODEOWNERS
.gitignore
.stylish-haskell.yaml
LICENSE
NOTICE
README.md
cabal.project
default.nix
nsh.sh
release.nix
scripts/prolog
shell.nix
```

Most seem fine, but what are these? Should we just delete them now?

```
.stylish-haskell.yaml
nsh.sh
scripts/prolog
```

@deepfire -- you still want `nsh.sh` here?

**Comment by nfrisby** [comment]
I included the `consensus-superowners` Team, but I don't think that's effective until that Team is added to this repository. That requires an Admin of this repository and possibly someone who is also a Maintainer of that Team (the team is "secret", so you might have to add the repo to the Team's Repositories page instead of vice versa).

We'll also need an admin of this repo to tick the "require code owner Approval" checkbox on the Branches rule for `master` in this repository's Settings page.

File: .github/CODEOWNERS
Comment (coot) [inline .github/CODEOWNERS:52]:
```suggestion
cabal.project.local.ci.windows     @coot @input-output-hk/devops @input-output-hk/consensus-superowners
```
I wrote that file :)

Comment (nfrisby) [reply .github/CODEOWNERS]: Fixed, thanks. I left devops in because of the lifting John Ky has been doing.

Feel free to remove devops if you think we should.

Actually, we could create another "secret" team devops-secret if we want them to have access without receiving all the notifications /shrug.

File: .github/CODEOWNERS
Comment (nfrisby) [inline .github/CODEOWNERS:52]: Add `network-superowners` to these four too?

Comment (coot) [reply .github/CODEOWNERS:52]: Maybe we could have just one secrete team for that? In `network-superowners` there's Duncan, Philip and I (only because I didn't know how to remove myself).  Is there anybody else in `consensus-superowners? Maybe Edsko.

Comment (nfrisby) [reply .github/CODEOWNERS:52]: Yeah; I'm not in it anymore (so it apparently is possible to remove yourself) but Edsko is.

My intuition is that we should have both groups and just pay the duplication price on some of the shared lines in this file. That seems the most future-compatible.

Comment (coot) [reply .github/CODEOWNERS:52]: :+1:

**Review by nfrisby — APPROVED** [review]
LGTM! Thanks!

The only remaining item that I'm not certain is already done is what I mentioned in https://github.com/input-output-hk/ouroboros-network/pull/2894#issuecomment-769321841 --- someone (admin, so maybe Wessels? maybe Coutts? I don't have visibility and can't otherwise remember) needs to configure this repository so that the "superowners" groups have access to it.

**Comment by coot** [comment]
bors merge

**Comment by coot** [comment]
bors merge

**Pull request merged**
